docker/Dockerfile

```diff
@@ -16,7 +16,7 @@
 # under the License.
 #
 
-ARG BASE_ALPINE_IMAGE="alpine:3.13"
+ARG BASE_ALPINE_IMAGE="alpine:3.16"
```

Comment:
Why not 3.17.1, the current latest?

Reply:
because alpine 3.17 uses gcc 12, which would not match the title of this PR

we could update this and other CI jobs to gcc 12 (except for centos 7, which doesn't have a `devtoolset-12` yet)

but maybe do that in a separate PR?